packages/@aws-cdk/aws-stepfunctions-tasks/lib/emrcontainers/delete-virtual-cluster.ts

```diff
@@ -0,0 +1,73 @@
+import * as iam from '@aws-cdk/aws-iam';
+import * as sfn from '@aws-cdk/aws-stepfunctions';
+import * as cdk from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { integrationResourceArn, validatePatternSupported } from '../private/task-utils';
+
+/**
+ * Properties to define a EMR Containers DeleteVirtualCluster Task
+ */
+export interface EmrContainersDeleteVirtualClusterProps extends sfn.TaskStateBaseProps {
+
+  /**
+   * The ID of the virtual cluster that will be deleted.
+   */
+  readonly virtualClusterId: sfn.TaskInput;
+}
+
+/**
+ * Deletes a EMR Containers virtual cluster as a Task.
+ *
+ * @see https://docs.amazonaws.cn/en_us/step-functions/latest/dg/connect-emr-eks.html
+ */
+export class EmrContainersDeleteVirtualCluster extends sfn.TaskStateBase {
+
+  private static readonly SUPPORTED_INTEGRATION_PATTERNS: sfn.IntegrationPattern[] = [
+    sfn.IntegrationPattern.REQUEST_RESPONSE,
+    sfn.IntegrationPattern.RUN_JOB,
+  ];
+
+  protected readonly taskMetrics?: sfn.TaskMetricsConfig;
+  protected readonly taskPolicies?: iam.PolicyStatement[];
+
+  private readonly integrationPattern: sfn.IntegrationPattern;
+
+  constructor(scope: Construct, id: string, private readonly props: EmrContainersDeleteVirtualClusterProps) {
+    super(scope, id, props);
+    this.integrationPattern = props.integrationPattern ?? sfn.IntegrationPattern.REQUEST_RESPONSE;
+
+    validatePatternSupported(this.integrationPattern, EmrContainersDeleteVirtualCluster.SUPPORTED_INTEGRATION_PATTERNS);
+
+    this.taskPolicies = this.createPolicyStatements();
+  }
+
+  /**
+   * @internal
+   */
+  protected _renderTask(): any {
+    return {
+      Resource: integrationResourceArn('emr-containers', 'deleteVirtualCluster', this.integrationPattern),
+      Parameters: sfn.FieldUtils.renderObject({
+        Id: this.props.virtualClusterId.value,
+      }),
+    };
+  };
+
+  private createPolicyStatements(): iam.PolicyStatement[] {
+    const actions = ['emr-containers:DeleteVirtualCluster'];
+    if (this.integrationPattern === sfn.IntegrationPattern.RUN_JOB) {
+      actions.push('emr-containers:DescribeVirtualCluster');
+    }
+
+    return [new iam.PolicyStatement({
+      resources: [
+        cdk.Stack.of(this).formatArn({
+          service: 'emr-containers',
+          resource: '/virtualclusters',
+          resourceName: sfn.JsonPath.isEncodedJsonPath(this.props.virtualClusterId.value) ? '*' : this.props.virtualClusterId.value,
+        }),
```

Comment:
these should still be using the arnFormat parameter to add the `/`. I find it hard to believe that some actions require the slash and some do not

Reply:
Surprisingly, when I ran the integ tests without the ``/`` I was getting a problem with the task not executing because of an iam policy issue.

Reply:
This is for both delete virtual cluster and start job run.

Reply:
The `/` should always exist when synthesized, it just depends on whether you put it in to the `resource` string or `formatArn` does it for you by specifying an `arnFormat`